### ISSUE TITLE
Nested Types (only one level)

### ISSUE THREAD

**Issue by jezzsantos** [TNG/ArchUnitNET#109]
I have some classes like this because this is a pattern we use in our codebases for various good reasons:
```
    public static class Events
    {
        public static class Thing
        {
            public class CreateThing : IChangeEvent
            {
                public string EntityId { get; set; }

                public DateTime ModifiedUtc { get; set; }
            }
        }
    }
```

Now, I want to capture all the classes that derive from the interface `IChangeEvent` but as it turns out, I never see any.

This statement captures the `Thing` class, but not the `CreateThing` class.
```
var classes = ArchRuleDefinition.Types(true).That().ResideInNamespace("MyNamespace", true);
```

The framework seems to support nested classes, but it only seems to support one level of nesting. 

So this does not work either:

```
var classes = ArchRuleDefinition.Types(true).That().ResideInNamespace("MyNamespace", true).And().AreNested();

```

How can I capture my deeper nested classes, no matter how nested they are? 
(max 3 levels is probably all we need in practice)

**Comment by fgather** [comment]
Hi @jezzsantos ,

thanks for reporting, I think this is a bug. We'll try to handle it.

**Comment by brandhuf** [comment]
Fixed in #118

**Comment by fgather** [comment]
Fix was released in 0.8.1
